src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -154,20 +176,40 @@ protected virtual void ReadSecurityTokenServiceTypeRoleDescriptor(WsFederationCo
             // Process <RoleDescriptorr>
             reader.ReadStartElement();
 
+            // Warn if there is no KeyDescriptor element for Signing
+            // Throw if there is no SecurityTokenEndpoint element
+            var hasKeyDescriptorForSigning = false;
+            var hasSecurityTokenEndpoint = false;
+
             while (reader.IsStartElement())
             {
                 if (IsKeyDescriptorForSigning(reader))
+                {
+                    hasKeyDescriptorForSigning = true;
                     ReadKeyDescriptorForSigning(configuration, reader);
+                }
                 else if (reader.IsStartElement(WsFederationConstants.Elements.SecurityTokenEndpoint, WsFederationConstants.Namespaces.FederationNamespace))
+                {
+                    hasSecurityTokenEndpoint = true;
                     ReadSecurityTokenEndpoint(configuration, reader);
+                }
                 else if (reader.IsStartElement())
+                {
                     reader.ReadOuterXml();
+                }
                 else
+                {
                     throw XmlUtil.LogReadException(LogMessages.IDX13003, reader.Name);
+                }
             }
 
             // Process </RoleDescriptorr>
             reader.ReadEndElement();
+
+            if (!hasKeyDescriptorForSigning)
```

Comment:
We don't need these flags, we can check the configuration object.
In both cases just log warnings.
I know we discussed throwing if no token endpoint. But let's just log.